### PR TITLE
drivers: sensor: mpu9150: Initial driver support

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
This might not make sense, but to me it kind of does.
I add this here as an alternative way forward with the AK8975 driver.

### CODE REVIEW

File: dts/bindings/sensor/invensense,mpu9150.yaml [hunk 1]
```diff
@@ -9,6 +9,12 @@ compatible: "invensense,mpu9150"
 
 include: i2c-device.yaml
 
+properties:
```
Comment (galak) [inline dts/bindings/sensor/invensense,mpu9150.yaml:12]: we can't introduce a new property here, have to do this w/o that.

**Review by galak — CHANGES_REQUESTED** [review]
DTS binding for mpu9150 can't introduce new properties.  Needs to match "upstream" linux binding.

https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/iio/gyroscope/invensense,mpu3050.yaml

**Comment by benjaminbjornsson** [comment]
> DTS binding for mpu9150 can't introduce new properties. Needs to match "upstream" linux binding.
> 
> https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/iio/gyroscope/invensense,mpu3050.yaml

If the binding can’t change, then my feeling is that the only sensible option left is to have a private attribute for enabling/disabling pass-through.

**Comment by benjaminbjornsson** [comment]
Fixed here: #47701

**Pull request closed without merging**